blocks/library/latest-posts/index.js

```diff
@@ -36,28 +36,6 @@ registerBlockType( 'core/latest-posts', {
 
 	keywords: [ __( 'recent posts' ) ],
 
-	attributes: {
```

Comment:
I've been thinking about this a bit more. This kind of feels odd to me—that opening the block JS file has no trace of attributes. (We would also need to handle the categories block.)

Reply:
We could have the block implementer include attributes here, but it's redundant information that could easily fall out of sync. 

A single source of truth would be ideal, but if we want to support server validation of block attributes, this would mean that single source must live on the server (or in some way compatible*). In some ways, this could even be seen as a good thing, but it also fragments the implementation of a block (some in the server, some in the client) and hinders the ease of implementing a block which can live exclusively in the client.

_*_ Some previous discussion of compatible formats such as block manifest (JSON) file but... that doesn't really eliminate fragmentation as much as it introduces more 😄

Reply:
Well, if we move all config to a generic json file, and separate edit / save into their own exported `edit.js` `save.js` (or a common `index.js` file), it might be ok.

The source of truth of a block seems like it should be independent from server/client distinction.

Reply:
Related: https://github.com/WordPress/gutenberg/pull/1905#issuecomment-319092720

Recreating matchers in an environment-agnostic setting is a challenge. This isn't as much of a problem for server-rendered blocks because we assume we don't need matchers (unclear yet if this is a fair assumption).